src/gui/search/searchwidget.cpp

```diff
@@ -161,6 +162,12 @@ bool SearchWidget::eventFilter(QObject *object, QEvent *event)
         {
             closeTab(tabIndex);
             return true;
+        } else if (mouseEvent->button() == Qt::RightButton)
+        {
+            QMenu m;
+            m.addAction(tr("Close tab"), [=](){ closeTab(tabIndex); return true; });
```

Comment:
Use this variation: https://doc.qt.io/qt-5/qmenu.html#addAction-5 with `context = this`.